build/scraper.mjs

```diff
@@ -13,47 +13,51 @@ import WarframeScraper from './wikia/scrapers/WarframeScraper.mjs';
 import VersionScraper from './wikia/scrapers/VersionScraper.mjs';
 import readJson from './readJson.mjs';
 
+import {get, getJSON} from './network.mjs';
+
 const locales = await readJson(new URL('../config/locales.json', import.meta.url));
 
 const prod = process.env.NODE_ENV === 'production';
-// eslint-disable-next-line no-control-regex
-const sanitize = (str) => str.replace(/\\r|\r?\n|\x09/g, '').replace(/\\\\"/g, "'");
-const agent = process.env.SOCK5_HOST
-  ? new Agent({
-      socksHost: process.env.SOCKS5_HOST,
-      socksPort: process.env.SOCKS5_PORT,
-      socksUsername: process.env.SOCKS5_USER,
-      socksPassword: process.env.SOCKS5_PASS,
-    })
-  : undefined;
-const get = async (url, disableProxy = !prod, compress = false) => {
-  const res = await fetch(url, {
-    agent: disableProxy ? undefined : agent,
-  });
-  return compress === false ? Uint8Array.from(await res.buffer()) : res.text();
-};
-const getJSON = async (url, disableProxy) => JSON.parse(sanitize(await get(url, disableProxy, true)));
 
 /**
  * Retrieves the base item data necessary for the parsing process
  */
 class Scraper {
+  endpointCache = new Map();
   /**
    * Get Endpoints from Warframe's origin file
    * @param {boolean} [manifest] to fetch only the manifest or everything else
    * @param {string} [locale] Locale to fetch data for
    */
   async fetchEndpoints(manifest, locale) {
-    const origin = `https://content.warframe.com/PublicExport/index_${locale || 'en'}.txt.lzma`;
-    const raw = await get(origin, !prod);
-    const decompressed = lzma.decompress(raw);
-    const manifestRegex = /(\r?\n)?ExportManifest.*/;
 
-    // We either don't need the manifest, or *only* the manifest
-    if (manifest) {
-      return manifestRegex.exec(decompressed)[0].replace(/\r?\n/, '');
-    }
-    return decompressed.replace(manifestRegex, '').split(/\r?\n/g);
+    let attemptsLeft = 5;
+    while(attemptsLeft > 0){
+      try{        
+        const origin = `https://origin.warframe.com/PublicExport/index_${locale || 'en'}.txt.lzma`;
+        
+        let raw = this.endpointCache.get(origin);
+        if(raw == undefined){
+          raw = await get(origin, !prod); 
+        }
+  
+        const decompressed = lzma.decompress(raw);        
+        this.endpointCache.set(origin, raw); //Cache endpoint only if lzma.decrompress didn't throw an error (valid file)
+
+        const manifestRegex = /(\r?\n)?ExportManifest.*/;  
+        // We either don't need the manifest, or *only* the manifest
+        if (manifest) {
+          return manifestRegex.exec(decompressed)[0].replace(/\r?\n/, '');
+        }
+        return decompressed.replace(manifestRegex, '').split(/\r?\n/g);
+      }catch (error){      
+        if(attemptsLeft>0){
+          attemptsLeft--;
+        }else{
+          throw error;
+        }
+      }
+    }    
```

Comment:
let's move the wrapping logic for this into a function in network, i.e. as a callback fn param on `get` or something similar